plugins/functions/string.c

```diff
@@ -190,6 +249,10 @@ func_string_substring(grn_ctx *ctx, int n_args, grn_obj **args,
     GRN_TEXT_SET(ctx, substring, start, end - start);
   }
 
+  if(GRN_TEXT_LEN(substring) == 0 && default_value){
```

Comment:
Does `string_substring(string, -1, -1, {"default_value": "XXX"})` work?
It seems that it returns `"XXX"` not an empty string.

Reply:
OK. I was wrong.

I think that an empty string is better for the 0 `length` case.